app/jobs/caldav/sync_event_job.rb

```diff
@@ -0,0 +1,47 @@
+module Caldav
+  class SyncEventJob < ApplicationJob
+    include GoodJob::ActiveJobExtensions::Concurrency
+
+    good_job_control_concurrency_with(
+      perform_limit: 1,
+      key: -> { "Caldav::SyncEventJob-#{arguments.first}" }
+    )
+
+    def perform(agents_rdv_id, agent_id, caldav_event_url: nil)
+      @agents_rdv_id = agents_rdv_id
+      @agent_id = agent_id
+
+      return unless agent
+
+      Sentry.set_user({ id: agent.id, role: "Agent", email: agent.email })
+
+      return unless agent.caldav_configured?
+
+      if caldav_event_url.present?
+        if agents_rdv
+          ics = IcalFormatters::Ics.from_payload(agents_rdv.rdv.payload(:update, agents_rdv.agent)).to_ical
+          agent.caldav_client.events.update(caldav_event_url, ics)
+        else
+          agent.caldav_client.events.delete(caldav_event_url)
+        end
+      else
+        ics = IcalFormatters::Ics.from_payload(agents_rdv.rdv.payload(:create, agents_rdv.agent)).to_ical
+        identifier = "agents_rdv-#{agents_rdv.id}.ics"
+        event = agent.caldav_client.events.create(agent.caldav_agenda_url, identifier, ics)
+        # Le provider Caldav n’utilise pas forcément l’identifiant qu’on lui donne pour créer l’event
+        # on stocke donc l’url complète de l’event créé pour être sûr de pouvoir le retrouver.
+        agents_rdv.update!(caldav_url: event.url)
```

Comment:
J'ai lié mon compte sur Alain Mairie, qui a un seul RDV, et j'ai observé les jobs dans le dashboard GoodJob. J'ai constaté qu'il y en avait 3 jobs `Caldav::SyncEventJob` à la file !

La solution : 


```suggestion
        # Aussi, évite de lancer les callbacks définis sur AgentsRdv,
        # en particulier celui qui est à l'origine de l'exécution de ce job.
        agents_rdv.update_columns(caldav_url: event.url)
```

Cela permet d'éviter d'appeler les callbacks du modèle `AgentsRdv` ici, sinon on va re-enqueue un job ! On a déjà traité ce problème pour Outlook dans `app/jobs/outlook/sync_event_job.rb:70`. 😉 

Note : puisque `caldav_url` est stable (pas modifiée dans les jobs d'update), on ne tombe pas dans une boucle infinie.

Reply:
Bien vu effectivement, en plus je l’ai déjà fait à un autre endroit 😬